crates/task-impls/src/view_sync.rs

```diff
@@ -452,8 +409,31 @@ impl<
                                 )
                                 .await;
                         }
-                        if let Some((_key, relay_task_info)) =
-                            self.relay_task_map.remove_entry(&TYPES::Time::new(i))
+                        if let Some((_key, relay_task_info)) = self
```

Comment:
Nit: Is it possible to combine these `if let` checks? Looks like they are sending the direct message in the same way, though associated with different phases. Maybe we could use `||` to send at most one `Shutdown`.